Fix projects

Make sure that all projects are using dotnet 8 and that all references between projects are working.

You didn't update it in both workflow files. The builds also fails with
```
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/ValidationPipeline.cs(156,10): error CS1513: } expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/PluginManager.cs(465,10): error CS1002: ; expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/ValidationPipeline.cs(161,10): error CS1513: } expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/ValidationPipeline.cs(162,26): error CS1026: ) expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/ValidationPipeline.cs(162,26): error CS1002: ; expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/PluginManager.cs(465,10): error CS1513: } expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/ValidationPipeline.cs(162,28): error CS1002: ; expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/ValidationPipeline.cs(162,28): error CS1513: } expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Monitoring/ValidationPipelineDashboard.cs(65,48): error CS1002: ; expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Monitoring/ValidationPipelineDashboard.cs(65,48): error CS1513: } expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Monitoring/ValidationPipelineDashboard.cs(66,42): error CS1002: ; expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Monitoring/ValidationPipelineDashboard.cs(66,42): error CS1513: } expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Monitoring/ValidationPipelineDashboard.cs(67,62): error CS1002: ; expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Monitoring/ValidationPipelineDashboard.cs(67,62): error CS1513: } expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Monitoring/ValidationPipelineDashboard.cs(68,48): error CS1002: ; expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Monitoring/ValidationPipelineDashboard.cs(68,48): error CS1513: } expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Monitoring/ValidationPipelineDashboard.cs(69,92): error CS1002: ; expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Monitoring/ValidationPipelineDashboard.cs(69,92): error CS1513: } expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Monitoring/ValidationPipelineDashboard.cs(70,76): error CS1002: ; expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Analytics/TimeSeriesStorage.cs(207,9): error CS1524: Expected catch or finally [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Analytics/TimeSeriesStorage.cs(209,9): error CS0106: The modifier 'public' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Analytics/TimeSeriesStorage.cs(271,9): error CS0106: The modifier 'public' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Analytics/TimeSeriesStorage.cs(284,9): error CS0106: The modifier 'public' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/Analytics/TimeSeriesStorage.cs(317,10): error CS1513: } expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(75,5): error CS0106: The modifier 'public' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(75,30): error CS1001: Identifier expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(88,9): error CS0106: The modifier 'private' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(96,5): error CS0106: The modifier 'public' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(113,5): error CS0106: The modifier 'public' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(137,5): error CS0106: The modifier 'public' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(146,5): error CS0106: The modifier 'private' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(187,5): error CS0106: The modifier 'private' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(200,5): error CS0106: The modifier 'private' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(213,5): error CS0106: The modifier 'private' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(245,5): error CS0106: The modifier 'private' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(254,5): error CS0106: The modifier 'private' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(271,5): error CS0106: The modifier 'private' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(312,5): error CS0106: The modifier 'public' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(330,5): error CS0106: The modifier 'public' is not valid for this item [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
Error: /home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/Implementation/CodeValidation/PerformanceMonitor.cs(336,2): error CS1513: } expected [/home/runner/work/CodeBuddy/CodeBuddy/CodeBuddy.Core/CodeBuddy.Core.csproj]
```

--- a/.github/workflows/build-cli.yml
+++ b/.github/workflows/build-cli.yml
@@ -1,0 +1,34 @@
+name: Build CLI
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 8.0.x
+        
+    - name: Restore dependencies
+      run: dotnet restore
+      
+    - name: Build
+      run: dotnet build --no-restore --configuration Release
+      
+    - name: Build CLI
+      run: dotnet publish CodeBuddy.CLI/CodeBuddy.CLI.csproj -c Release -o publish
+      
+    - name: Upload CLI artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: codebuddy-cli
+        path: publish/

--- a/.github/workflows/build-cli.yml
+++ b/.github/workflows/build-cli.yml
@@ -22,9 +22,9 @@ jobs:
       run: dotnet restore
       
     - name: Build
-      run: dotnet build --no-restore --configuration Release
+      run: dotnet build --no-restore
       
-    - name: Build CLI
+    - name: Build CLI Release
       run: dotnet publish CodeBuddy.CLI/CodeBuddy.CLI.csproj -c Release -o publish
       
     - name: Upload CLI artifact

--- a/.github/workflows/build-cli.yml
+++ b/.github/workflows/build-cli.yml
@@ -28,7 +28,9 @@ jobs:
       run: dotnet publish CodeBuddy.CLI/CodeBuddy.CLI.csproj -c Release -o publish
       
     - name: Upload CLI artifact
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
         name: codebuddy-cli
         path: publish/
+        if-no-files-found: error
+        compression-level: 6

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -1,0 +1,34 @@
+name: Build CLI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 8.0.x
+        
+    - name: Restore dependencies
+      run: dotnet restore
+      
+    - name: Build
+      run: dotnet build --no-restore
+      
+    - name: Build CLI
+      run: dotnet publish CodeBuddy.CLI/CodeBuddy.CLI.csproj -c Release -o publish
+
+    - name: Upload CLI artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: codebuddy-cli
+        path: publish/

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -28,7 +28,9 @@ jobs:
       run: dotnet publish CodeBuddy.CLI/CodeBuddy.CLI.csproj -c Release -o publish
 
     - name: Upload CLI artifact
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
         name: codebuddy-cli
         path: publish/
+        if-no-files-found: error
+        compression-level: 6

--- a/CodeBuddy.CLI/CodeBuddy.CLI.csproj
+++ b/CodeBuddy.CLI/CodeBuddy.CLI.csproj
@@ -2,7 +2,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
   </PropertyGroup>

--- a/CodeBuddy.E2ETests/CodeBuddy.E2ETests.csproj
+++ b/CodeBuddy.E2ETests/CodeBuddy.E2ETests.csproj
@@ -1,16 +1,16 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <IsPackable>false</IsPackable>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.1.0" />
-    <PackageReference Include="xunit" Version="2.4.1" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3" />
-    <PackageReference Include="Moq" Version="4.18.1" />
-    <PackageReference Include="FluentAssertions" Version="6.7.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
+    <PackageReference Include="xunit" Version="2.6.3" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.5.5" />
+    <PackageReference Include="Moq" Version="4.20.70" />
+    <PackageReference Include="FluentAssertions" Version="6.12.0" />
   </ItemGroup>
 
   <ItemGroup>

--- a/CodeBuddy.IntegrationTests/CodeBuddy.IntegrationTests.csproj
+++ b/CodeBuddy.IntegrationTests/CodeBuddy.IntegrationTests.csproj
@@ -1,19 +1,19 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <IsPackable>false</IsPackable>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.1.0" />
-    <PackageReference Include="xunit" Version="2.4.1" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3">
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
+    <PackageReference Include="xunit" Version="2.6.3" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.5.5">
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
-    <PackageReference Include="Moq" Version="4.18.1" />
-    <PackageReference Include="FluentAssertions" Version="6.7.0" />
+    <PackageReference Include="Moq" Version="4.20.70" />
+    <PackageReference Include="FluentAssertions" Version="6.12.0" />
   </ItemGroup>
 
   <ItemGroup>

--- a/CodeBuddy.Tests/CodeBuddy.Tests.csproj
+++ b/CodeBuddy.Tests/CodeBuddy.Tests.csproj
@@ -1,21 +1,21 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
     <IsPackable>false</IsPackable>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.1.0" />
-    <PackageReference Include="Moq" Version="4.18.1" />
-    <PackageReference Include="xunit" Version="2.4.1" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3">
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.8.0" />
+    <PackageReference Include="Moq" Version="4.20.70" />
+    <PackageReference Include="xunit" Version="2.6.3" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.5.5">
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
-    <PackageReference Include="coverlet.collector" Version="3.1.2">
+    <PackageReference Include="coverlet.collector" Version="6.0.0">
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>

